crates/weaver_semconv/data/event.yaml

```diff
@@ -0,0 +1,181 @@
+groups:
+  - id: device.app.lifecycle
+    stability: experimental
+    type: event
+    name: device.app.lifecycle
+    brief: >
+      This event represents an occurrence of a lifecycle transition on Android or iOS platform.
+    note: >
+      This event identifies the fields that are common to all lifecycle events for android and iOS using
+      the `android.state` and `ios.state` fields. The `android.state` and `ios.state` attributes are
+      mutually exclusive.
+    body:
+      id: device_lifecycle
+      type: map
+      requirement_level: optional
+      fields:
+        - id: ios.state
+          stability: experimental
+          requirement_level:
+            conditionally_required: if and only if `os.name` is `ios`
+          note: >
+            The iOS lifecycle states are defined in the [UIApplicationDelegate documentation](https://developer.apple.com/documentation/uikit/uiapplicationdelegate#1656902),
+            and from which the `OS terminology` column values are derived.
+          brief: >
+            This attribute represents the state the application has transitioned into at the occurrence of the event.
+          type: enum
```

Comment:
This is the `enum` example the main difference between an `any_value` enum type and the `AttributeType` enum type is

- you need to provide the "type" name (ie. `enum`)
- The `allow_custom_values` and `members` are now at the same "level" (not indented)

If this was the same as an AttributeType enum it would look like (which is a little problematic)
```yaml
brief: >
            This attribute represents the state the application has transitioned into at the occurrence of the event.
          type:
            allow_custom_values: false
            members:
              - id: active
                value: 'active'
                brief: >
                  The app has become `active`. Associated with UIKit notification `applicationDidBecomeActive`.
              - id: inactive
                value: 'inactive'
                brief: >
                  The app is now `inactive`. Associated with UIKit notification `applicationWillResignActive`.
             # ... Pruned ...
```